bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -386,12 +412,37 @@ public BufferedReadChannel getFromChannels(long logId) {
      *
      * @return least unflushed log id.
      */
-    synchronized long getLeastUnflushedLogId() {
-        return leastUnflushedLogId;
+    long getLeastUnflushedLogId() {
+        return recentlyCreatedEntryLogsStatus.getLeastUnflushedLogId();
     }
 
-    synchronized long getCurrentLogId() {
-        return logChannel.getLogId();
+    long getPreviousAllocatedEntryLogId() {
+        return entryLoggerAllocator.getPreallocatedLogId();
+    }
+
+    boolean rollLogsIfEntryLogLimitReached() throws IOException {
```

Comment:
I think this method should belong to EntryLogManager. so you don't need to expose `getCopyOfCurrentLogs()` to EntryLogger. for the simple entry log manager, you don't need to covert it back to a HashSet and do the while loop here.